src/NUnitFramework/framework/Constraints/DelayedConstraint.cs

```diff
@@ -35,9 +35,129 @@ namespace NUnit.Framework.Constraints
     ///</summary>
     public class DelayedConstraint : PrefixConstraint
     {
+        /// <summary>
+        /// 
+        /// </summary>
+        public class DelayedConstraint3 : DelayedConstraint
+        {
+            private DelayedConstraint1 _parent;
+
+            /// <summary>
+            /// 
+            /// </summary>
+            /// <param name="parent"></param>
+            public DelayedConstraint3(DelayedConstraint1 parent)
+                : base(parent.BaseConstraint, parent._delayInMilliseconds, parent._pollingInterval)
+            {
+                _parent = parent; ;
+            }
+
+            /// <summary>
+            /// 
+            /// </summary>
+            public DelayedConstraint Minute
+            {
+                get
+                {
+                    return _parent;
+                }
+            }
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        public class DelayedConstraint2 : DelayedConstraint
+        {
+            private readonly DelayedConstraint1 _parent;
+
+            /// <summary>
+            /// 
+            /// </summary>
+            /// <param name="parent"></param>
+            /// <exception cref="NotImplementedException"></exception>
+            public DelayedConstraint2(DelayedConstraint1 parent)
+                : base(parent.BaseConstraint, parent._delayInMilliseconds, parent._pollingInterval)
+            {
+                _parent = parent;
+                _delayInterval = parent._delayInterval;
+            }
+
+            /// <summary>
+            /// 
+            /// </summary>
+            /// <param name="milliSeconds"></param>
+            /// <returns></returns>
+            public DelayedConstraint3 PollEvery(int milliSeconds)
+            {
+                _parent._pollingInterval = milliSeconds;
+                return new DelayedConstraint3(_parent);
+            }
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        public class DelayedConstraint1 : DelayedConstraint
+        {
+            ///<summary>
+            /// Creates a new DelayedConstraint
+            ///</summary>
+            ///<param name="baseConstraint">The inner constraint to decorate</param>
+            ///<param name="delayInMilliseconds">The time interval after which the match is performed</param>
+            ///<exception cref="InvalidOperationException">If the value of <paramref name="delayInMilliseconds"/> is less than 0</exception>
+            public DelayedConstraint1(IConstraint baseConstraint, int delayInMilliseconds) : base(baseConstraint, delayInMilliseconds) { }
```

Comment:
For this constructor I have 2 options. If I kept it in the new nested class then I wouldn't have to change tests like the one here to call the new class.

```
[TestFixture]
	public class AsyncDelayedConstraintTests
	{
		[Test]
		public void ConstraintSuccess()
		{
			Assert.IsTrue(new DelayedConstraint(new EqualConstraint(1), 100)
				.ApplyTo(async () => await One()).IsSuccess);
		}

```

Also have to change the After method in Constraint.cs to this,

```
/// <summary>
        /// Returns a DelayedConstraint with the specified delay time.
        /// </summary>
        /// <param name="delayInMilliseconds">The delay in milliseconds.</param>
        /// <returns></returns>
        public DelayedConstraint.DelayedConstraint1 After(int delayInMilliseconds)
        {
            return new DelayedConstraint.DelayedConstraint1(new DelayedConstraint( 
                Builder == null ? this : Builder.Resolve(),
                delayInMilliseconds));
        }
```

But thinking about it I realize if I give the nested class a better name it would still make sense to make this change.  This would also make the nested class depend on the parent which would be consistent. I'll change this and see how it looks first.